src/Domain.LinnApps/Stock/StoresPallet.cs

```diff
@@ -4,18 +4,146 @@
 
     public class StoresPallet
     {
-        public int PalletNumber { get; set; }
+        public StoresPallet()
+        {
+        }
+
+        public StoresPallet(
+            int palletNumber,
+            string description,
+            StorageLocation locationId,
+            DateTime? dateInvalid,
+            DateTime? dateLastAudited,
+            string accessible,
+            string storesKittable,
+            int? storesKittablePriority,
+            string salesKittable,
+            int? salesKittablePriority,
+            DateTime? allocQueueTime,
+            LocationType locationType,
+            int? auditedBy,
+            StockPool defaultStockPool,
+            string stockType,
+            string stockState,
+            int? auditOwnerId,
+            int? auditFrequencyWeeks,
+            string auditedByDepartmentCode,
+            string mixStates,
+            int? cage)
+        {
+            this.PalletNumber = palletNumber;
+            this.Description = description;
+            this.LocationIdCode = locationId.LocationId;
+            this.DateInvalid = dateInvalid;
+            this.DateLastAudited = dateLastAudited;
+            this.Accessible = accessible;
+            this.StoresKittable = storesKittable;
+            this.StoresKittablePriority = storesKittablePriority;
+            this.SalesKittable = salesKittable;
+            this.SalesKittablePriority = salesKittablePriority;
+            this.AllocQueueTime = allocQueueTime;
+            this.LocationTypeId = locationType?.Code;
+            this.AuditedBy = auditedBy;
+            this.DefaultStockPoolId = defaultStockPool?.StockPoolCode;
+            this.StockType = stockType;
+            this.StockState = stockState;
+            this.AuditOwnerId = auditOwnerId;
+            this.AuditFrequencyWeeks = auditFrequencyWeeks;
+            this.AuditedByDepartmentCode = auditedByDepartmentCode;
+            this.MixStates = mixStates;
+            this.Cage = cage;
+        }
 
-        public int LocationId { get; set; }
+        public int PalletNumber { get; set; }
 
         public string Description { get; set; }
 
+        public int LocationIdCode { get; set; }
+
+        public StorageLocation LocationId { get; set; }
```

Comment:
should probably name this as more like a storage location rather than a storage location id